x/evm/keeper/vote_handler.go

```diff
@@ -0,0 +1,158 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/utils/slices"
+
+	"github.com/axelarnetwork/axelar-core/x/evm/types"
+	nexus "github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	vote "github.com/axelarnetwork/axelar-core/x/vote/exported"
+)
+
+// NewVoteHandler returns the handler for processing vote delivered by the vote module
+func NewVoteHandler(cdc codec.Codec, keeper types.BaseKeeper, nexus types.Nexus) vote.VoteHandler {
+	return func(ctx sdk.Context, result *vote.Vote) error {
+		events, err := types.UnpackEvents(cdc, result.Results)
+		if err != nil {
+			return err
+		}
+
+		if len(events) == 0 {
+			return nil
```

Comment:
Should technically panic here due to the check at https://github.com/axelarnetwork/axelar-core/blob/7f9b90bbd3fe95729766b04769d1f946a6d55097/x/vote/keeper/msg_server.go#L75. Return error at least.